src/client/Microsoft.Identity.Client/ManagedIdentity/ServiceFabricManagedIdentitySource.cs

```diff
@@ -21,11 +21,33 @@ internal class ServiceFabricManagedIdentitySource : AbstractManagedIdentity
 
         public static AbstractManagedIdentity Create(RequestContext requestContext)
         {
+            Uri endpointUri;
             string identityEndpoint = EnvironmentVariables.IdentityEndpoint;
 
             requestContext.Logger.Info(() => "[Managed Identity] Service fabric managed identity is available.");
 
-            if (!Uri.TryCreate(identityEndpoint, UriKind.Absolute, out Uri endpointUri))
+            if (requestContext.ServiceBundle.Config.IsFmiServiceFabric)
```

Comment:
I was considering letting MSAL autodetect the correct env variable, but I am not sure if it is a good idea. Both env variables may be set (IDENTITY_ENDPOINT and APP_IDENTITY_ENDPOINT), so I don't want to introduce bugs. Adding an api for this that will only be used from MISE will make this more robust.

Reply:
Let's take a step back and think how we'd go about doing this in MSAL if we were to productize getting the FMI credential. 

The FMI credential is just a token with a specific audience. And it can use different endpoints. 

Can we use the existing MSI APIs, and have different logic based on the audience? i.e. if app developer requests token for `api://AzureFMITokenExchange` (or the GUID format) - then detect it?

Reply:
I understand what you are saying, but since this scenario isnt intended to use a resource and is simply acquiring the token from MITS, all of the additional application logic for resources/scopes/caching is not needed. I would also would be concerned if someone wanted to use MI to exchange an FMI credential for an access token using `api://AzureFMITokenExchange`, but instead they just get another FMI Credential.  so, maybe instead of using the fmi token exchange resource, we can create some string like "FmiMitsAcquisition" or something so that no one will trigger this logic in MSAL by accident? But if you feel like no one will ever use `api://AzureFMITokenExchange` with a MI app, this this will work.

Reply:
I'm ok with an experimental API `mi,AcquireFmiCredential()`

Reply:
@trwalke - can you put toghter a small design doc for this and let's discuss with the team?